pkg/engine/lifecycle_test.go

```diff
@@ -4334,3 +4345,208 @@ func TestImportUpdatedID(t *testing.T) {
 		}
 	}
 }
+
+func TestDeleteTarget(t *testing.T) {
+	names := []string{"A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L"}
+
+	// Try refreshing a stack with combinations of the above resources as target to destroy.
+	subsets := combinations.All(names)
+
+	for _, subset := range subsets {
```

Comment:
Might be nice to have a few handcrafted subsets here to cover particularly interesting cases.